File: ExampleApp/app/src/main/java/com/motiv/example/dao/MetaDao.java
package com.motiv.example.dao;

import androidx.room.*;
import com.motiv.example.Meta;

@Dao
public interface MetaDao {

    @Insert
    public long saveMeta(com.motiv.example.Meta meta);

    @Delete
    public int deleteMeta(com.motiv.example.Meta meta);

    @Query("SELECT * FROM meta")
    public java.util.List<Meta> getMetas();

    @Query("SELECT * FROM meta WHERE id = :id")
    public Meta getMeta(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveMetas(java.util.List<com.motiv.example.Meta> meta);
}

File: ExampleApp/app/src/main/java/com/motiv/example/dao/MetaDao.kt
 
package com.motiv.example.dao
import androidx.room.*
import com.motiv.example.Meta
import kotlin.collections.List
@Dao
public interface MetaDao {

    @Insert
    public fun saveMeta(meta: com.motiv.example.Meta): Long@Delete
    public fun deleteMeta(meta: com.motiv.example.Meta): Int@Query("SELECT * FROM meta")
    public fun getMetas(): List<Meta>@Query("SELECT * FROM meta WHERE id = :id")
    public fun getMeta(id: String): Meta@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveMetas(meta: List<com.motiv.example.Meta>): Unit
}
